src/server/components/dashboard/controllers.ts

```diff
@@ -152,12 +157,66 @@ export async function listsController(req: Request, res: Response, next: NextFun
       isNewUser,
       lists,
       csrfToken: getCSRFToken(req),
+      dashboardBoxes: calculateDashboardBoxes(lists as ListsForDashboard[], !req.user?.isAdministrator),
     });
   } catch (error) {
     next(error);
   }
 }
 
+export function calculateSortOrder(
+  queryParamSortOrder: Prisma.ListsForDashboardOrderByWithRelationInput
+): Array<Record<string, string>> {
+  const defaultSortOrder = {
+    country: "asc",
+    type: "asc",
+  };
+
+  // TODO: Object.hasOwn is recommended but is not currently supported by tsc.
+  // eslint-disable-next-line no-prototype-builtins
+  if (!queryParamSortOrder.hasOwnProperty("admins")) {
+    return Object.entries(defaultSortOrder).map(convertEntryToObject);
+  }
+
+  const newSortOrder = {
+    ...queryParamSortOrder,
+    ...defaultSortOrder,
+  };
+
+  return Object.entries(newSortOrder).map(convertEntryToObject);
```

Comment:
I don't think this check on admin is necessary. if queryParamSortOrder is empty or undefined object, it won't affect sort order during the spread operation 

```suggestion
 
  const sortOrder = {
    ...defaultSortOrder, // you want this first, because in future country or type could be sorted the other way.
    ...queryParamSortOrder
  };

  return Object.entries(newSortOrder).map(convertEntryToObject);
```

Reply:
sure, my worry is that if someone types a value that isn't recognised by prisma orderBy in the url they'll mess the query up so this is to protect against that. But yes, it's not really necessary.